VirtoCommerce.QuoteModule.Data/Services/QuoteRequestServiceImpl.cs

```diff
@@ -160,14 +164,16 @@ public void Delete(string[] ids)
         {
             using (var repository = _repositoryFactory())
             {
+                var changedEntries = new List<GenericChangedEntry<QuoteRequest>>();
                 var dbQuotes = repository.GetQuoteRequestByIds(ids);
-                var quotes = GetByIds(ids);
                 foreach (var dbQuote in dbQuotes)
                 {
-                    _eventPublisher.Publish(new QuoteRequestChangeEvent(Platform.Core.Common.EntryState.Deleted, quotes.First(x => x.Id == dbQuote.Id), null));
+                    changedEntries.Add(new GenericChangedEntry<QuoteRequest>(dbQuote.ToCoreModel(), EntryState.Deleted));
                     repository.Remove(dbQuote);
                 }
+                _eventPublisher.Publish(new QuoteRequestChangeEvent(changedEntries));
                 repository.UnitOfWork.Commit();
+                _eventPublisher.Publish(new QuoteRequestChangeEvent(changedEntries));
```

Comment:
Why do we need to call the same QuoteRequestChangeEvent twice? Probably it makes sense in case of -Changing and -Changed events but do not know why we need to do it here.